runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5802,9 +5802,7 @@ class INTERPRETER_CLASS
 
 		VMStructHasBeenUpdated(REGISTER_ARGS);
 
-		if (immediateAsyncPending()) {
```

Comment:
Would moving the async check into a `JDK24 #ifdef block` result in a loss of functionality for other JDK versions?

Reply:
I don't think there is any possibility pre-JDK24 that needs the asyncCheck as `Continuation.yield()` shouldn't need async before completing the next bytecode.

Reply:
I meant loss of functionality for other async events that are not dependent on `Continuation.yield()`.

Reply:
Instead of removing this async check, you probably just need to add the new one to cover `Continuation.yield()`?

Reply:
@gacholio Could confirm if it is safe to remove this async check?

Reply:
This isn't a normal async check - it's the immediate version (thing that must be processed before running any more code on the thread). There is currently one immediate - pop frames (which can't happen here because we're in a native), so I think this can safely be removed.